src/Middleware/passingNodeData.ts

```diff
@@ -1,18 +1,27 @@
 import express, {Request} from "express";
 import {NodeService} from "../Services/NodeService";
+import {Node} from "../Models/Node";
+import logger from "../Services/Logger";
 
 export async function passNodeData(
     req: express.Request,
     res: express.Response,
     next: express.NextFunction
 ) {
-    const {url, address} = req.body;
-
-    const node = await NodeService.getNodeByData(url, address);
-    if (node) {
-        res.locals.node = node;
+    try {
+        const node = await Node.findByPk(req.params.nodeId)
```

Comment:
Shouldn't you check first if you have the nodeId or url&address? You don't have both and this will fail in the methods that provide only url and address.

Reply:
if (req.params && req.params.nodeId) {
const node = await Node.findByPk(req.params.nodeId)
} else if (req.body.url && req.body.address) {
}